doc/user/content/sql/create-source/webhook.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "CREATE SOURCE: Webhook"
+description: "Ingesting data into Materialize with HTTP requests"
+pagerank: 50
+menu:
+  main:
+    parent: 'create-source'
+    identifier: webhook
+    name: Webhook
+    weight: 40
+---
+
+{{< private-preview />}}
+
+{{% create-source/intro %}}
+Webhook sources expose a URL that allow your other applications to push data into Materialize.
+{{% /create-source/intro %}}
+
+
+## Syntax
+
+{{< diagram "create-source-webhook.svg" >}}
+
+## `webhook_body_format`
+
+{{< diagram "webhook-body-format.svg" >}}
+
+## `webhook_check_option`
+
+{{< diagram "webhook-check-option.svg" >}}
+
+Field                            | Use
+---------------------------------|--------------------------
+  _src_name_                     | The name for the source
+ **IN CLUSTER** _cluster_name_   | The [cluster](/sql/create-cluster) to maintain this source.
+ **INCLUDE HEADERS**             | Include a column named `'headers'` of type `map[text => text]` containing the headers of the request.
+ **CHECK**                       | Specify a boolean expression that is used to validate each request received by the source.
+
+### `BODY FORMAT` options
+
+Field    | Value    | Description
+---------|----------|-------------------------
+`TEXT`   | `text`   | Parses the body of a request as UTF-8 text. If the body is not valid UTF-8, a response of `400` Bad Request will be returned.
+`JSON`   | `jsonb`  | Parses the body of a request as JSON. If the body is not valid JSON, a respose of `400` Bad Request will be returned.
+`BYTES`  | `bytea`  | Does no parsing of the request, stores the body as it was received.
+
+### `CHECK WITH` options
+
+Field                  | Value               | Description
+-----------------------|---------------------|--------------
+`BODY`                 | `text` or `bytea`   | Provide a column `'body'` to the check expression. The column can be renamed with the optional **AS** _alias_ statement, and the type can be changed to `bytea` with the optional **BYTES** keyword.
+`HEADERS`              | `map[text=>text]` or `map[text=>bytea]` | Provide a column `'headers'` to the check expression. The column can be renamed with the optional **AS** _alias_ statement, and the type can be changed to `map[text => bytea]` with the optional **BYTES** keyword.
+`SECRET` _secret_name_ | `text` or `bytea`    | Provide a column _secret_name_ to the check expression, with the value of the [`SECRET`](/sql/create-secret) _secret_name_. The column can be renamed with the optional **AS** _alias_ statement, and the type can be changed to `bytea` with the optional **BYTES** keyword.
+
+## Webhook URL
```

Comment:
This should document that the URL is open to the public internet and has no security.

Reply:
Done!